promql/prettier/node.go

```diff
@@ -0,0 +1,163 @@
+package prettier
+
+import (
+	"github.com/prometheus/prometheus/promql/parser"
+	"reflect"
+)
+
+type nodeInfo struct {
+	head parser.Expr
+	// Node details.
+	columnLimit int
+	history     []reflect.Type
+	item        parser.Item
+}
+
+func newNodeInfo(head parser.Expr, columnLimit int, item parser.Item) *nodeInfo {
+	return &nodeInfo{
+		head:        head,
+		columnLimit: columnLimit,
+		item:        item,
+	}
+}
+
+func (p *nodeInfo) violatesColumnLimit() bool {
+	return len(p.head.String()) > p.columnLimit
+}
+
+// getNode returns the node corresponding to the given position range in the AST.
+func (p *nodeInfo) getNode(root parser.Expr, item parser.Item) parser.Expr {
+	history, node, _ := p.nodeHistory(root, item.PositionRange(), []reflect.Type{})
+	p.history = history
+	return node
+}
+
+func (p *nodeInfo) baseIndent(item parser.Item) int {
+	history, _, _ := p.nodeHistory(p.head, item.PositionRange(), []reflect.Type{})
+	history = reduceContinuous(history, "*parser.BinaryExpr")
+	return len(history)
+}
+
+// contains verifies whether the current node contains a particular entity.
+func (p *nodeInfo) contains(element string) bool {
```

Comment:
`element` should be an enum.

Reply:
Or just write two functions.

Reply:
Going with `enum`.

Reply:
Addressed in recent commits. Hence, resolving.